pkg/controller/support/credentials/credentials.go

```diff
@@ -98,16 +106,17 @@ func Persist(
 			}
 
 			helpMessage = append(helpMessage, "basic authentication username and password")
-		case ptr.Deref(cred.CloudProvided, false):
+		default:
 			pac, err := CloudAuthRegistry.RetrieveAuthorization(ctx, logger, cred.Server)
 			if err != nil {
-				return "", nil, fmt.Errorf("cloud registry authorization failed: %w", err)
+				if err != cloudauth.ErrNoLoader {
+					return "", nil, fmt.Errorf("cloud registry authorization failed: %w", err)
+				}
+				return "", nil, fmt.Errorf("credential %v is missing auth section", cred)
```

Comment:
I guess I was making the assumption:
* Passing no basic or secret auth means you want to use creds provided by the cloud
* If we can't load credentials for the cloud, you must fill in the auth section

Is there better wording you'd recommend for the "cloud registry authorization failed" message?

Reply:
A different way to look at it is to just have the wrapped error case.
If instead we have either ErrNoLoader have a meaningful message or return a different error like 'No credentials for server %s found' would make this less specialized.
Because the missing auth section is really something the cloudauth package knows but not the credentials.
Never realized that this nesting is probably just wrong. credentials.cloudauth but whatever.